### PR TITLE
Created a method to hide image from public when issue is hidden

### LINKED ISSUE #1104: When an issue is hidden, make sure that it's corresponding image in the BLT hosted files is also hidden and not public

### PR DESCRIPTION
**Pull request by pranav-leo**
fix #1104

### CODE REVIEW

**Comment by pranav-iitr** [comment]
@fredfalcon pls, review the changes

**Review by fredfalcon — CHANGES_REQUESTED** [review]
Can you please setup a test to make sure this works?

**Review by fredfalcon — CHANGES_REQUESTED** [review]
Please add a test

**Comment by pranav-iitr** [comment]
> Please add a test

is a test evolving only models is sufficient or a selenium based test is required?

**Comment by pranav-iitr** [comment]
@fredfalcon  pls check

**Comment by pranav-iitr** [comment]
made the changes pls check

**Comment by pranav-iitr** [comment]
> The last 6 lines of the test can be rewritten to assert true (file exists)

Made the changes please check

**Comment by pranav-iitr** [comment]
> Please add a test

made changes pls check

**Comment by pranav-iitr** [comment]
> > Please add a test
> 
> made changes pls check

@fredfalcon

**Review by fredfalcon — CHANGES_REQUESTED** [review]
Please take out the try / except and unchange any code not related to the change

File: website/models.py [hunk 1]
```diff
@@ -278,10 +294,40 @@ def get_absolute_url(self):
...
 class IssueScreenshot(models.Model):
     image = models.ImageField(upload_to="screenshots", validators=[validate_image])
     issue = models.ForeignKey(Issue,on_delete=models.CASCADE,related_name="screenshots")
+    hide = models.BooleanField(default=False)
+
+@receiver(post_save, sender=Issue)
+def update_issue_image_access(sender, instance, **kwargs):
+    print(sender,instance)
+    
+    blob = go
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/models.py]: removed this bug

**Comment by pranav-iitr** [comment]
> Please take out the try / except and unchange any code not related to the change

rectified

**Review by fredfalcon — CHANGES_REQUESTED** [review]
Please remove the field and clean up any unused print or code

File: website/models.py [hunk 2]
```diff
@@ -278,10 +284,41 @@ def get_absolute_url(self):
...
 class IssueScreenshot(models.Model):
     image = models.ImageField(upload_to="screenshots", validators=[validate_image])
     issue = models.ForeignKey(Issue,on_delete=models.CASCADE,related_name="screenshots")
+    hide = models.BooleanField(default=False)
+
+@receiver(post_save, sender=Issue)
+def update_issue_image_access(sender, instance, **kwargs):
+    print(sender,instance)
+    
+    # blob = go
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/models.py]: removed pls check

File: website/migrations/0073_issuescreenshot_hide.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.1 on 2023-03-11 13:42
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/migrations/0073_issuescreenshot_hide.py]: removed pls check

**Review by fredfalcon — CHANGES_REQUESTED** [review]
Please see inline

File: website/tests.py [hunk 4]
```diff
@@ -71,16 +79,41 @@ def test_post_bug(self):
...
+        issue_screenshot_list_orignal = IssueScreenshot.objects.filter(
+            issue=Test_Object.id)
+
+        Test_Object.is_hidden = True
+        Test_Object.save()
+        for screenshot in issue_screenshot_list_orignal:
+            filename = screenshot.image.name
+
+            if default_storage.exists(filename):
+                self.assertTrue(False, "files exist")
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/tests.py:103]: resolved pls check

File: website/models.py [hunk 5]
```diff
@@ -21,6 +21,10 @@
 from mdeditor.fields import MDTextField
 from decimal import Decimal
 from captcha.fields import CaptchaField
+from django.core.files.storage import default_storage
+import uuid
+
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/models.py:25]: resolved

File: website/models.py [hunk 6]
```diff
@@ -29,6 +33,8 @@ def create_auth_token(sender, instance=None, created=False, **kwargs):
         Wallet.objects.create(user=instance)
 
 
+
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/models.py]: resoloved

File: website/models.py [hunk 7]
```diff
@@ -278,11 +284,40 @@ def get_absolute_url(self):
...
+                    extension = filename.split(".")[-1] 
+                    name = filename[12:99]+"hidden" + str(uuid.uuid4()) + "." + extension
+                    default_storage.save(f"screenshots/{name}",screenshot.image)   
+                    default_storage.delete(old_name)
+                    screenshot.image=f"screenshots/{name}"
+                    screenshot.image.name=f"screenshots/{name}"
+                    screenshot.save()  
+                    
+
+                    # if not settings.DEBUG:
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/models.py]: resoloved

File: website/tests.py [hunk 8]
```diff
@@ -6,27 +17,21 @@
...
-from webdriver_manager.chrome import ChromeDriverManager
-from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
 
 class MySeleniumTests(LiveServerTestCase):
     fixtures = ['initial_data.json']
 
     @classmethod
     def setUpClass(cls):
-        print (sys.path)
+        print(sys.path)
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/tests.py]: resolved

File: website/tests.py [hunk 9]
```diff
@@ -38,9 +43,12 @@ def tearDownClass(cls):
     def test_signup(self):
         self.selenium.get('%s%s' % (self.live_server_url, '/accounts/signup/'))
         self.selenium.find_element("name", "username").send_keys('bugbugbug')
-        self.selenium.find_element("name", "email").send_keys('bugbugbug@bugbug.com')
-        self.selenium.find_element("name", "password1").send_keys('6:}jga,6mRKNUqMQ')
-        self.selenium.find_element("name", "password2").send_keys('6:}jga,6mRKNUqMQ')
+        self.selenium.find_element(
```
[the remark this answers is not in the document]
Comment (pranav-iitr) [reply website/tests.py]: resolved

**Review by fredfalcon — DISMISSED** [review]
Looking better.  Can you please remove any changes that are not directly related to your change? Because it will make it easier to review.

**Comment by pranav-iitr** [comment]
> Looking better. Can you please remove any changes that are not directly related to your change? Because it will make it easier to review.

have removed

**Comment by pranav-iitr** [comment]
> Looking better. Can you please remove any changes that are not directly related to your change? Because it will make it easier to review.

removed @fredfalcon  pls check

**Comment by AtmegaBuzz** [comment]
LGTM! @fredfalcon

**Comment by pranav-iitr** [comment]
@fredfalcon pls merge if everything is fine

**Review by fredfalcon — APPROVED** [review]

**Pull request merged**